### ISSUE TITLE
AttentionBias and relative positional encoding

### ISSUE THREAD

**Issue by YoelShoshan** [facebookresearch/xformers#925]
I'm converting [hf transformers T5](https://github.com/huggingface/transformers/blob/main/src/transformers/models/t5/modeling_t5.py#L453) to use [memory_efficient_attention](https://facebookresearch.github.io/xformers/components/ops.html)()

I've reached a point that I'm getting identical results between the original implementation and when using `memory_efficient_attention()`, however, I pass `attn_bias` as a Tensor, which I'm aware isn't optimized and results in FLASH attention not being selected.

So I started trying to convert the code to use [AttentionBias](https://github.com/facebookresearch/xformers/blob/main/xformers/ops/fmha/attn_bias.py) 

In the original T5 code, the default positional embedding is relative, and it is injected into the attention mechanism by adding it to the same place that `attn_bias` is being added (on Q@K output, before the softmax)
However, it's not clear if it's possible to support this while using AttentionBias inherited class (or creating a new one)
**Any help on this will be highly appreciated :)**

Note - I understand that there are alternative positional encoding methods, but I need the code to support existing model weights as well, which do rely on that specific positional encoding method.

**Comment by danthe3rd** [comment]
Hi,
This is not something supported efficiently by xFormers at the moment. You will need to create a custom kernel to do that. It might be similar to what was done for SAM
https://pytorch.org/blog/accelerating-generative-ai/ (See section "Triton: Custom SDPA for fused relative positional encoding")

**Comment by IceClear** [comment]
@YoelShoshan Hi, have you solved it? I also have similar issues.
BTW, I just wonder will it be very difficult to support custom attn_bias? @danthe3rd 
Thanks in advance.

**Comment by danthe3rd** [comment]
We have no plans to support it for now
